File: week-07/day-2/simba/src/main/java/com/bankofsimba/simba/model/BankAccount.java
package com.bankofsimba.simba.model;

public class BankAccount {
    private String name;
    private double balance;
    private String animalType;
    private static int index = 0;
    private int id;
    private boolean isKing;

    public BankAccount(String name, double balance, String animalType) {
        this.name = name;
        this.balance = balance;
        this.animalType = animalType;
        this.id = index++;
        isKing = false;
    }

    public String getName() {
        return name;
    }

    public double getBalance() {
        return balance;
    }

    public String getAnimalType() {
        return animalType;
    }

    public int getId() {
        return id;
    }

    public boolean isKing() {
        return isKing;
    }

    public void setKing() {
        isKing = true;
    }
}

File: week-11/day-1/Simba/Simba/Models/BankAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Simba.Models
{
    public class BankAccount 
    {
        public string Name { get; set; }
        public long Balance { get; set; }
        public AnimalType AnimalType { get; set; }
        public int Id { get; private set; }
        public bool IsKing { get; set; }
        public bool IsGood { get; set; }
    
        private static int IdGenerator;

        public BankAccount()
        {
            IdGenerator++;
        }

        public BankAccount(string name, long balance, AnimalType animalType)
        {
            Name = name;
            Balance = balance;
            AnimalType = animalType;
            IdGenerator++;
            Id = IdGenerator;

        }

        public BankAccount(string name, long balance, AnimalType animalType, bool isKing)
        {
            Name = name;
            Balance = balance;
            AnimalType = animalType;
            IdGenerator++;
            Id = IdGenerator;
            IsKing = isKing;
        }

        public BankAccount(string name, long balance, AnimalType animalType, bool isKing, bool isGood)
        {
            Name = name;
            Balance = balance;
            AnimalType = animalType;
            IdGenerator++;
            Id = IdGenerator;
            IsGood = isGood;
        }

        public void IncreaseBalance()
        {
            Balance += 100;
        }
    }
}
